### PR TITLE
[PRCA][Technical] Refactor the ReportProcessor to a module

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

**Comment by eyalantzMS** [comment]
Try providing some language reference next time, to simplify reviewing. i.e. http://www.howtogeek.com/203778/how-scopes-affect-powershell-scripts 

LGTM

File: Tasks/SonarQubePostTest/PRCA/ReportProcessor-Module.psm1 [hunk 1]
```diff
@@ -1,6 +1,8 @@
-$ProjectGuidAndFilePathMap = @{}
-$ComponentKeyAndPathMap = @{}
-$ComponentKeyAndRelativePathCache = @{}
+$script:ProjectGuidAndFilePathMap = @{}
```
Comment (bgavrilMS) [inline Tasks/SonarQubePostTest/PRCA/ReportProcessor-Module.psm1]: As far as I can tell, the script scope will prevent these variables from being accessed from outside the module.

File: Tasks/SonarQubePostTest/PRCA/ReportProcessor-Module.psm1 [hunk 2]
```diff
@@ -231,4 +233,7 @@ function AnnotateIssuesWithRelativePath
     return $issues
 }
 
-#endregion
\ No newline at end of file
+#endregion
+
+# Export the public functions 
+Export-ModuleMember -Function 'FetchAnnotatedNewIssues'
```
Comment (bgavrilMS) [inline Tasks/SonarQubePostTest/PRCA/ReportProcessor-Module.psm1]: One has to explicitly export the functions. This is the equivalent of public / private visibility.

File: Tests/L0/SonarQubePostTest/PRCA/ReportProcessorTests.ps1 [hunk 3]
```diff
@@ -1,9 +1,10 @@
 [CmdletBinding()]
 param()
 
+Import-Module -Name "$PSScriptRoot\..\..\..\..\Tasks\SonarQubePostTest\PRCA\ReportProcessor-Module.psm1" -Verbose
+Import-Module -Name "$PSScriptRoot\..\..\..\..\Tasks\SonarQubePostTest\PRCA\PostComments-Module.psm1" -Verbose
```
Comment (bgavrilMS) [inline Tests/L0/SonarQubePostTest/PRCA/ReportProcessorTests.ps1]: I have already transformed this script to a module based on feedback on an earlier PR

**Pull request merged**